kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSFunctionDeclarationImpl.kt

```diff
@@ -41,15 +40,11 @@ class KSFunctionDeclarationImpl private constructor(internal val ktFunctionSymbo
             cache.getOrPut(ktFunctionSymbol) { KSFunctionDeclarationImpl(ktFunctionSymbol) }
     }
 
-    // TODO: upstream deprecated `KtSymbolKind` in favor of `KaSymbolLocation` but they are
-    // semantically different as the new location does not suggest if the function is a lambda or not.
-    // This is a breaking change in upstream deprecation.
```

Comment:
So the comment here mentions that `KtSymbolKind` and `KtSymbolLocation` are not the same but the implementation of `symbolKind` is:
```kotlin
public val symbolKind: KaSymbolKind
    get() = when (location) {
        KaSymbolLocation.TOP_LEVEL -> KaSymbolKind.TOP_LEVEL
        KaSymbolLocation.CLASS -> KaSymbolKind.CLASS_MEMBER
        KaSymbolLocation.PROPERTY -> KaSymbolKind.ACCESSOR
        KaSymbolLocation.LOCAL -> KaSymbolKind.LOCAL
    }
```